kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,657 @@
+
+    /**
+     * @brief This sets the level of echo for the solving strategy
+     * @param Level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     */
+    virtual void SetEchoLevel(const int Level)
+    {
+        mEchoLevel = Level;
+    }
+
+    /**
+     * @brief This returns the level of echo for the solving strategy
+     * @details
+     * {
+     * 0 -> Mute... no echo at all
+     * 1 -> Printing time and basic informations
+     * 2 -> Printing advanced information
+     * 3 -> Print of debug informations: Echo of stiffness matrix, Dx, b...
+     * }
+     * @return Level of echo for the solving strategy
+     */
+    virtual int GetEchoLevel()
+    {
+        return mEchoLevel;
+    }
+
+    /**
+     * This sets the build level
+     * @param Level The build level
+     * @details
+     * {
+     * 0 -> Set up the DOF set just once
+     * 1 -> Set up the DOF set at the beginning of each solution step
+     * }
+     */
+    virtual void SetRebuildLevel(int Level)
+    {
+        mRebuildLevel = Level;
+    }
+
+    /**
+     * @brief This returns the build level
+     * @details
+     * {
+     * 0 -> Set up the DOF set just once
+     * 1 -> Set up the DOF set at the beginning of each solution step
+     * }
+     * @return The build level
+     */
+    virtual int GetRebuildLevel()
+    {
+        return mRebuildLevel;
+    }
+
+    /**
+     * @brief This function sets the flag that says if the mesh is moved
+     * @param Flag True if the mesh is moved, false otherwise
+     */
+    void SetMoveMeshFlag(bool Flag)
+    {
+        mMoveMeshFlag = Flag;
+    }
+
+    /**
+     * @brief This function returns the flag that says if the mesh is moved
+     * @return True if the mesh is moved, false otherwise
+     */
+    bool MoveMeshFlag()
+    {
+        return mMoveMeshFlag;
+    }
+
+    /**
+     * @brief This function is designed to move the mesh
+     * @note Be careful it just consider displacements, derive this method to adapt to your own strategies (ALE, FSI, etc...)
+     */
+    virtual void MoveMesh()
+    {
+        KRATOS_TRY
+
+        auto& r_nodes_array = GetModelPart().Nodes();
+        block_for_each(
+            r_nodes_array,
+            [](Node<3>& rNode){
+                noalias(rNode.Coordinates()) = rNode.GetInitialPosition().Coordinates();
+                noalias(rNode.Coordinates()) += rNode.FastGetSolutionStepValue(DISPLACEMENT);
+            }
+        );
+
+        KRATOS_INFO_IF("ExplicitSolvingStrategy", this->GetEchoLevel() > 0) << "Mesh moved." << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Operations to get the pointer to the model
+     * @return mrModelPart: The model part member variable
+     */
+    inline ModelPart& GetModelPart()
```

Comment:
```suggestion
    ModelPart& GetModelPart()
```

I would remove, the compiler anyway does what it wants

btw missing `const` version

Reply:
Done.